source/rrRoadRunner.cpp

```diff
@@ -7186,6 +7184,32 @@ namespace rr {
         }
     }
 
+    void RoadRunner::setSeed(long int seed, bool resetModel) {
+        Config::setValue(Config::RANDOM_SEED, seed);
+        if (resetModel) {
+            regenerateModel(true);
+            reset(SelectionRecord::TIME |
+            SelectionRecord::RATE |
+            SelectionRecord::FLOATING |
+            SelectionRecord::BOUNDARY |
+            SelectionRecord::COMPARTMENT |
+            SelectionRecord::GLOBAL_PARAMETER);
+        }
+        else {
+            impl->model->setRandomSeed(seed);
+            if (getIntegrator()->getName() == "gillespie")
```

Comment:
Instead of checking the current integrator, I think we need to find the gillepie integrator (if it exists) and set the seed.

Reply:
i.e. getIntegrator("gillespie"), and then call setSeed on it if it exists.